### ISSUE TITLE
Incorrect alarm status

### ISSUE THREAD

**Issue by peros550** [dustindclark/homebridge-envisalink#90]
Sorry to bother you again Dustin. 

Recently, I ve noticed a couple of times already the alarm system status in HomeKit is not in sync with the actual status. More specifically, I check the status remotely through EyezON and find the alarm as armed to away. However in HomeKit it shows off. Any idea?

**Comment by dustindclark** [comment]
Which version are you on? I saw something similar and put a fix in a patch version or 2 ago.

Dustin

**Comment by peros550** [comment]
I have the latest 1.0.4

**Comment by dustindclark** [comment]
Is it reproducible? Any logs?

Dustin

**Comment by peros550** [comment]
No idea how to reproduce it. I’ll be alerted and if/when happens again,  I will gather any logs and will surely post again. Meanwhile, would you prefer to keep this issue open or closed ?

**Comment by dustindclark** [comment]
Definitely leave it open. I’ll work to see if I can recreate as well.

**Comment by jclark16** [comment]
I just updated to 1.0.4 and noticed this happened when my alarm armed for "stay" mode (through an automation).  It armed as expected and then immediately said "alarm off".  This is the first time I've tried arming through Homebridge since updating to 1.0.4 - I can't recall which version I was on before but it was pre-1.0.1.  I'm unsure if the Zone 005 is related to the alarm status changing - it is the motion sensor in the room I was in.  Also, I don't usually bypass a sensor so I'm not sure if that's related, but the sensor started acting up today and I haven't had a chance to mess with it yet.

[12/24/2021, 11:00:00 PM] [Envisalink] sendcommand 0311C5
[12/24/2021, 11:00:00 PM] [Envisalink] Command 031 Acknowledged
[12/24/2021, 11:00:01 PM] [Envisalink] Keypad LED State: 8B 
[12/24/2021, 11:00:03 PM] [Envisalink] Exit Delay of Partition 1 in Progress
[12/24/2021, 11:01:00 PM] [Envisalink] Keypad LED State: 8A 
[12/24/2021, 11:01:01 PM] [Envisalink] Partition 1 is armed with one or more zones bypassed
[12/24/2021, 11:01:01 PM] [Envisalink] Partition 11 is Armed
[12/24/2021, 11:01:02 PM] [Envisalink] Partition 1 has been armed by Quick Arm, Auto Arm, Keyswitch, DLS or Wireless Key
[12/24/2021, 11:01:19 PM] [Envisalink] Zone 005 is open
[12/24/2021, 11:01:20 PM] [Envisalink] Zone 005 is restored
[12/24/2021, 11:01:57 PM] [Envisalink] Zone 005 is open
[12/24/2021, 11:01:58 PM] [Envisalink] Zone 005 is restored

Eyezon does report the same status as the keypad itself, which is armed.

(edit) I sorted out the issue with the sensor that I had to bypass, and now things are working normally.  Status showing as expected.

**Comment by axelbrunger** [comment]
Related to this issue: after restarting HomeBridge, the alarm status reported in Apple HomeKit is not in sync with the actual alarm status. For example if the alarm is in Away mode, after HomeBridge restart, it is in Off mode.

**Comment by readylmay** [comment]
Few issues in status I am seeing in 1.0.4, 1.0.5:
-My alarm status is also not accurate.
-The alarm does report armed, and then immediately reports disarmed and the state of the alarm is not accurate.
-When the exit delay is in progress, the state of the alarm should be set to "arming" with the animation on the accessory. 
-These behaviours exist through automations, setting the alarm at the panel or setting the status on the HomeKit accessory in the home app. #
-This is reproducible everytime.
-I notice that the partition that is being reported as partition "11" - this is similar to an earlier comment. Likely a logging issue. 

-Its likely that one of the 3 last states that is being reporting by the panel that is falling through your switch statement and being caught by the default disarmed state in "bindSecurityPanel()".

Wish I could help further, more of a Objective C guy myself. 


This is a log when setting the alarm for stay with a few zones overrided at the panel:


```
[12/02/2022, 14:19:01] [Envisalink] Keypad LED Flash State: 08 
[12/02/2022, 14:19:03] [Envisalink] Keypad LED Flash State: 00 
[12/02/2022, 14:19:03] [Envisalink] Keypad LED State: 89 
[12/02/2022, 14:19:03] [Envisalink] Partition 1 Ready
[12/02/2022, 14:19:05] [Envisalink] Keypad LED State: 8B 
[12/02/2022, 14:19:07] [Envisalink] Exit Delay of Partition 1 in Progress
[12/02/2022, 14:19:50] [Envisalink] Keypad LED State: 8A 
[12/02/2022, 14:19:50] [Envisalink] Partition 1 is armed with one or more zones bypassed
[12/02/2022, 14:19:51] [Envisalink] Partition 11 is Armed
[12/02/2022, 14:19:52] [Envisalink] Partition 1 has been armed by Quick Arm, Auto Arm, Keyswitch, DLS or Wireless Key
```

**Comment by readylmay** [comment]
> I just updated to 1.0.4 and noticed this happened when my alarm armed for "stay" mode (through an automation). It armed as expected and then immediately said "alarm off". This is the first time I've tried arming through Homebridge since updating to 1.0.4 - I can't recall which version I was on before but it was pre-1.0.1. I'm unsure if the Zone 005 is related to the alarm status changing - it is the motion sensor in the room I was in. Also, I don't usually bypass a sensor so I'm not sure if that's related, but the sensor started acting up today and I haven't had a chance to mess with it yet.
> 
> [12/24/2021, 11:00:00 PM] [Envisalink] sendcommand 0311C5 [12/24/2021, 11:00:00 PM] [Envisalink] Command 031 Acknowledged [12/24/2021, 11:00:01 PM] [Envisalink] Keypad LED State: 8B [12/24/2021, 11:00:03 PM] [Envisalink] Exit Delay of Partition 1 in Progress [12/24/2021, 11:01:00 PM] [Envisalink] Keypad LED State: 8A [12/24/2021, 11:01:01 PM] [Envisalink] Partition 1 is armed with one or more zones bypassed [12/24/2021, 11:01:01 PM] [Envisalink] Partition 11 is Armed [12/24/2021, 11:01:02 PM] [Envisalink] Partition 1 has been armed by Quick Arm, Auto Arm, Keyswitch, DLS or Wireless Key [12/24/2021, 11:01:19 PM] [Envisalink] Zone 005 is open [12/24/2021, 11:01:20 PM] [Envisalink] Zone 005 is restored [12/24/2021, 11:01:57 PM] [Envisalink] Zone 005 is open [12/24/2021, 11:01:58 PM] [Envisalink] Zone 005 is restored
> 
> Eyezon does report the same status as the keypad itself, which is armed.
> 
> (edit) I sorted out the issue with the sensor that I had to bypass, and now things are working normally. Status showing as expected.

@jclark16 when you bypass a zone for stay or away, are the alarm state accurate? I cannot get an accurate alarm state if any zones are by passed.

**Comment by jclark16** [comment]
@readylmay no, if a zone is bypassed then it reverts alarm to state "off" in HomeKit, regardless of whether it is actually "away" or "stay".

I haven't sorted out how to efficiently debug or do development on Homebridge.  I see the module as debug level logging output which could give us more clues of what is happening, but I'm not certain how to make that active.  I already have the verbose logging enabled but I think it's just getting passed down to NodeAlarmProxy.

**Comment by readylmay** [comment]
Thanks @jclark16 - same behaviour here.

@dustindclark should we open up another issue, or is this sufficient with the logs with have. Let me know if you need anything else to help track the one down....


Edit - put Homebridge into debug mode.
The below sequence is me arming the alarm for stay with a zone overridden, waiting for the exit delay, having the alarm armed and then disarming. All done at the panel:


```
[18/02/2022, 15:18:45] [Envisalink] Keypad LED Flash State: 08 
[18/02/2022, 15:18:47] [Envisalink] Keypad LED Flash State: 00 
[18/02/2022, 15:18:47] [Envisalink] Keypad LED State: 89 
[18/02/2022, 15:18:47] [Envisalink] Partition 1 Ready
[18/02/2022, 15:18:47] [Envisalink] Inside partitionUpdate: {
  "evtType": "partitionUpdate",
  "partition": 1,
  "code": "650",
  "status": "ready"
}
[18/02/2022, 15:18:47] [Envisalink] Restoring existing partition accessory from cache: Alarm
[18/02/2022, 15:18:47] [Envisalink] Setting accessory details for zone: {
  "name": "Alarm",
  "number": 1,
  "enableChimeSwitch": false,
  "chimeCommand": "0711*4",
  "pin": "xxxx",
  "status": {
    "shortCode": "650",
    "longCode": "650",
    "description": "Partition Ready",
    "text": "ready",
    "action": "updatepartition",
    "verbSuffix": "Ready"
  }
}
[18/02/2022, 15:18:47] [Envisalink] Alarm Ready
[18/02/2022, 15:18:49] [Envisalink] Keypad LED State: 8B 
[18/02/2022, 15:18:51] [Envisalink] Exit Delay of Partition 1 in Progress
[18/02/2022, 15:18:51] [Envisalink] Inside partitionUpdate: {
  "evtType": "partitionUpdate",
  "partition": 1,
  "code": "656",
  "status": "exitdelay"
}
[18/02/2022, 15:18:51] [Envisalink] Restoring existing partition accessory from cache: Alarm
[18/02/2022, 15:18:51] [Envisalink] Setting accessory details for zone: {
  "name": "Alarm",
  "number": 1,
  "enableChimeSwitch": false,
  "chimeCommand": "0711*4",
  "pin": "xxxx",
  "status": {
    "shortCode": "656",
    "longCode": "656",
    "description": "Exit Delay in Progress",
    "text": "exitdelay",
    "action": "updatepartition",
    "verbSuffix": "in Progress"
  }
}
[18/02/2022, 15:18:51] [Envisalink] Alarm in Progress
[18/02/2022, 15:19:33] [Envisalink] Keypad LED State: 8A 
[18/02/2022, 15:19:34] [Envisalink] Partition 1 is armed with one or more zones bypassed
[18/02/2022, 15:19:34] [Envisalink] Inside partitionUpdate: {
  "evtType": "partitionUpdate",
  "partition": 1,
  "code": "702",
  "status": "partialclosing"
}
[18/02/2022, 15:19:34] [Envisalink] Restoring existing partition accessory from cache: Alarm
[18/02/2022, 15:19:34] [Envisalink] Setting accessory details for zone: {
  "name": "Alarm",
  "number": 1,
  "enableChimeSwitch": false,
  "chimeCommand": "0711*4",
  "pin": "xxxx",
  "status": {
    "shortCode": "702",
    "longCode": "702",
    "description": "Partial Closing",
    "text": "partialclosing",
    "action": "updatepartition",
    "verbSuffix": "is armed with one or more zones bypassed"
  }
}
[18/02/2022, 15:19:34] [Envisalink] Alarm is armed with one or more zones bypassed
[18/02/2022, 15:19:34] [Envisalink] Partition 11 is Armed
[18/02/2022, 15:19:34] [Envisalink] Inside partitionUpdate: {
  "evtType": "partitionUpdate",
  "partition": 1,
  "code": "652",
  "status": "armed",
  "armMode": "stay"
}
[18/02/2022, 15:19:34] [Envisalink] Restoring existing partition accessory from cache: Alarm
[18/02/2022, 15:19:34] [Envisalink] Setting accessory details for zone: {
  "name": "Alarm",
  "number": 1,
  "enableChimeSwitch": false,
  "chimeCommand": "0711*4",
  "pin": "xxxx",
  "status": {
    "shortCode": "652",
    "longCode": "652",
    "description": "Partition Armed",
    "text": "armed",
    "action": "updatepartition",
    "verbSuffix": "is Armed",
    "mode": "stay"
  }
}
[18/02/2022, 15:19:34] [Envisalink] System is armed. Mode: stay
[18/02/2022, 15:19:34] [Envisalink] Alarm is Armed
[18/02/2022, 15:19:35] [Envisalink] Partition 1 has been armed by Quick Arm, Auto Arm, Keyswitch, DLS or Wireless Key
[18/02/2022, 15:19:35] [Envisalink] Inside partitionUpdate: {
  "evtType": "partitionUpdate",
  "partition": 1,
  "code": "701",
  "status": "specialclosing",
  "armType": "specialClosing"
}
[18/02/2022, 15:19:35] [Envisalink] Restoring existing partition accessory from cache: Alarm
[18/02/2022, 15:19:35] [Envisalink] Setting accessory details for zone: {
  "name": "Alarm",
  "number": 1,
  "enableChimeSwitch": false,
  "chimeCommand": "0711*4",
  "pin": "xxxx",
  "status": {
    "shortCode": "701",
    "longCode": "701",
    "description": "Special Closing",
    "text": "specialclosing",
    "action": "updatepartition",
    "verbSuffix": "has been armed by Quick Arm, Auto Arm, Keyswitch, DLS or Wireless Key"
  }
}
[18/02/2022, 15:19:35] [Envisalink] Alarm has been armed by Quick Arm, Auto Arm, Keyswitch, DLS or Wireless Key
[18/02/2022, 15:19:42] [Envisalink] Keypad LED State: 80 
[18/02/2022, 15:19:43] [Envisalink] Partition Opened by User: 10040 
[18/02/2022, 15:19:43] [Envisalink] Inside dataUpdate: {
  "zone": {
    "1": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "61000128"
    },
    "2": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "61000229"
    },
    "3": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "6100032A"
    },
    "4": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "6100042B"
    },
    "5": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "6100052C"
    },
    "6": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "6100062D"
    },
    "7": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "6100072E"
    },
    "8": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "6100082F"
    },
    "9": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "61000930"
    },
    "10": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "61001028"
    },
    "11": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "61001129"
    },
    "12": {
      "send": "open",
      "name": "Zone Open",
      "code": "60901232"
    },
    "13": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "6100132B"
    },
    "14": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "6100142C"
    },
    "15": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "6100152D"
    },
    "16": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "6100162E"
    },
    "17": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "6100172F"
    },
    "18": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "61001830"
    },
    "19": {
      "send": "openrestore",
      "name": "Zone Restore",
      "code": "61001931"
    }
  },
  "partition": {
    "1": {
      "send": "useropening",
      "name": "User Opening",
      "code": "7501004091"
    }
  },
  "user": {
    "40": {
      "send": "useropening",
      "name": "User Opening",
      "code": "7501004091"
    }
  }
}
[18/02/2022, 15:19:43] [Envisalink] Restoring existing partition accessory from cache: Alarm
[18/02/2022, 15:19:43] [Envisalink] Setting accessory details for zone: {
  "name": "Alarm",
  "number": 1,
  "enableChimeSwitch": false,
  "chimeCommand": "0711*4",
  "pin": "xxxx",
  "status": {
    "shortCode": "750",
    "longCode": "7501004091",
    "description": "User Opening",
    "text": "useropening",
    "action": "updatepartition",
    "verbSuffix": ""
  }
}
[18/02/2022, 15:19:43] [Envisalink] Inside partitionUpdate: {
  "evtType": "partitionUpdate",
  "partition": 1,
  "code": "750",
  "status": "useropening",
  "armType": "userOpening",
  "userId": 40
}
[18/02/2022, 15:19:43] [Envisalink] Restoring existing partition accessory from cache: Alarm
[18/02/2022, 15:19:43] [Envisalink] Setting accessory details for zone: {
  "name": "Alarm",
  "number": 1,
  "enableChimeSwitch": false,
  "chimeCommand": "0711*4",
  "pin": "xxxx",
  "status": {
    "shortCode": "750",
    "longCode": "750",
    "description": "User Opening",
    "text": "useropening",
    "action": "updatepartition",
    "verbSuffix": ""
  }
}
[18/02/2022, 15:19:43] [Envisalink] Alarm 
[18/02/2022, 15:19:43] [Envisalink] Partition 1 has been Disarmed
[18/02/2022, 15:19:43] [Envisalink] Inside partitionUpdate: {
  "evtType": "partitionUpdate",
  "partition": 1,
  "code": "655",
  "status": "disarmed"
}
[18/02/2022, 15:19:43] [Envisalink] Restoring existing partition accessory from cache: Alarm
[18/02/2022, 15:19:43] [Envisalink] Setting accessory details for zone: {
  "name": "Alarm",
  "number": 1,
  "enableChimeSwitch": false,
  "chimeCommand": "0711*4",
  "pin": "xxxx",
  "status": {
    "shortCode": "655",
    "longCode": "655",
    "description": "Partition Disarmed",
    "text": "disarmed",
    "action": "updatepartition",
    "verbSuffix": "has been Disarmed"
  }
}
[18/02/2022, 15:19:43] [Envisalink] Alarm has been Disarmed
[18/02/2022, 15:19:47] [Envisalink] Partition 1 is NOT Ready
[18/02/2022, 15:19:47] [Envisalink] Inside partitionUpdate: {
  "evtType": "partitionUpdate",
  "partition": 1,
  "code": "651",
  "status": "notready"
}
[18/02/2022, 15:19:47] [Envisalink] Restoring existing partition accessory from cache: Alarm
[18/02/2022, 15:19:47] [Envisalink] Setting accessory details for zone: {
  "name": "Alarm",
  "number": 1,
  "enableChimeSwitch": false,
  "chimeCommand": "0711*4",
  "pin": "xxxx",
  "status": {
    "shortCode": "651",
    "longCode": "651",
    "description": "Partition Not Ready",
    "text": "notready",
    "action": "updatepartition",
    "verbSuffix": "is NOT Ready"
  }
}
[18/02/2022, 15:19:47] [Envisalink] Alarm is NOT Ready
```

**Comment by readylmay** [comment]
I believe the fix for this should be on line 78 of partitionAccessory.ts

case 'armedbypass' should be updated to catch the special closing (701) and partial closing (702) codes.

Update to:

`case 'armedbypass', 'partialclosing', 'specialclosing':`

**Comment by dustindclark** [comment]
Fixed in 1.0.6.
